### PR TITLE
Add neutron leakage postprocessors and allow for optional group constants

### LINKED ISSUE #234: Add postprocessor for calculating neutron leakage
Moltres currently does not have postprocessors which calculate the neutron leakage.

This issue can be closed when Moltres has postprocessors for calculating neutron leakage for each neutron group and the total across all groups through a given boundary, and the associated tests.

### PR DESCRIPTION
**Pull request by smpark7**
Fixes #234 and #167

Users can now choose which group constants they're providing to Moltres through the `group_constants` input parameter for `MoltresJsonMaterial`. This is useful for many problems which do not require every group constant type (e.g. eigenvalue problems do not require inverse neutron velocity data, neutronics-only problems do not require average fission energy released).

### CODE REVIEW

File: include/materials/NuclearMaterial.h [hunk 1]
```diff
@@ -64,6 +64,13 @@ class NuclearMaterial : public GenericConstantMaterial
 
   const VariableValue & _temperature;
 
+  // Number of neutron groups
+  unsigned int _num_groups;
```
Comment (LukeSeifert) [inline include/materials/NuclearMaterial.h:68]: Will this cause issues if a user tries to use more than 255 groups? I don't think it's common, but possibly worth considering for future-proofing.

File: include/postprocessors/NeutronLeakage.h [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include "SideIntegralVariablePostprocessor.h"
+
+/**
+ * This postprocessor computes the neutron leakage rate along a boundary
```
Comment (LukeSeifert) [inline include/postprocessors/NeutronLeakage.h:6]: For a single neutron group?

File: src/postprocessors/TotalNeutronLeakage.C [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+TotalNeutronLeakage::TotalNeutronLeakage(const InputParameters & parameters)
+  : SideIntegralPostprocessor(parameters),
+    ScalarTransportBase(parameters),
+    _vars(getCoupledMooseVars()),
+    _diffcoef(getMaterialProperty<std::vector<Real>>("diffcoef")),
+    _num_groups(getParam<unsigned int>("num_groups"))
+{
+  addMooseVariableDependency(_vars);
+  unsigned int n = coupledComponents("group_fluxes");
+  if (!(n == _num_groups))
```
Comment (LukeSeifert) [inline src/postprocessors/TotalNeutronLeakage.C:28]: How come you didn't use:
```suggestion
  if (n != _num_groups)
```

**Review by LukeSeifert — APPROVED** [review]
Looks good! I just had a few minor comments on things. I am not super familiar with C++, so if you want someone else to review I can wait to merge.

File: include/materials/NuclearMaterial.h (hunk 1 above)
Comment (smpark7) [reply include/materials/NuclearMaterial.h:68]: I think unsigned int goes up to 65k so it should be okay

File: include/postprocessors/NeutronLeakage.h (hunk 2 above)
Comment (smpark7) [reply include/postprocessors/NeutronLeakage.h:6]: Yea single. I'll make the change

File: src/postprocessors/TotalNeutronLeakage.C (hunk 3 above)
Comment (smpark7) [reply src/postprocessors/TotalNeutronLeakage.C:28]: Yea you're right. Good catch!

**Comment by smpark7** [comment]
Thanks @LukeSeifert! This is ready for another review. Feel free to merge if you approve

**Review by LukeSeifert — APPROVED** [review]
Looks good!

**Pull request merged**